### ISSUE TITLE
[UI] Header settings when creating new design should be dark mode compatible

### ISSUE THREAD

**Issue by Yashsharma1911** [meshery/meshery#7298]
#### Current Behavior
As you can see two issues in the below video
- settings header inside application section should contain `#00B39F` as background in dark mode
- when switching between dark mode and light mode text `settings` is displaced from its place

https://user-images.githubusercontent.com/71271069/223419340-d8877939-e827-4bad-bad4-769799038ffe.mov

#### Environment

- **Host OS:** Mac Linux Windows
- **Platform:** Docker or Kubernetes
- **Meshery Server Version:** stable-v
- **Meshery Client Version:** stable-v

---
#### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)
- 🛠 [Meshery Build & Release Strategy](https://docs.meshery.io/project/contributing/build-and-release)
- 📚 [Instructions for contributing to documentation](https://github.com/meshery/meshery/blob/master/CONTRIBUTING.md#documentation-contribution-flow)
   - Meshery documentation [site](https://docs.meshery.io/) and [source](https://github.com/meshery/meshery/tree/master/docs)
- 🎨 Wireframes and designs for Meshery UI in [Figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](https://discuss.layer5.io) and [Community Slack](http://slack.layer5.io)

**Comment by mangkoran** [comment]
Hello, may I work on this issue please?

**Comment by mangkoran** [comment]
Yes I am! I also able to reproduce the issue above.
![image](https://user-images.githubusercontent.com/57625126/223449183-0a2b50d7-45b6-4c47-8c1f-c96d352ba115.png)

**Comment by Yashsharma1911** [comment]
Thanks @mangkoran for showing

**Comment by mangkoran** [comment]
My findings so far:

1. "Settings" `AppBar` component background color is currently hard coded ([ref](https://github.com/meshery/meshery/blob/9fc8e0b7f60f3906965c7bbdf1996e264411f031/ui/components/MesheryMeshInterface/PatternServiceForm.js#L9)). Still WIP to make it respect light/dark mode theme (which is defined [here](https://github.com/meshery/meshery/blob/master/ui/themes/app.js) I suppose?)

**Comment by mangkoran** [comment]
For the `Settings` background color, is it supposed to have different color between light and dark theme? I could not find the design for dark theme in Figma.

![image](https://user-images.githubusercontent.com/57625126/223633551-2ec3e738-b2e9-48bd-a061-da031fdbeb05.png)

**Comment by mangkoran** [comment]
While investigating this ticket, I found another issue (although it seems it is unrelated to this ticket):

1. Whenever the `Application` `LazyPatternServiceForm` is opened, the CSS rule of `body` will changed, which changed `font`-related rules (visible on `> Designs > New Pattern` breadcrumb in video above).

**Comment by Yashsharma1911** [comment]
> My findings so far:
> 
> 1. `Settings` `AppBar` component background color is currently hard coded ([ref](https://github.com/meshery/meshery/blob/9fc8e0b7f60f3906965c7bbdf1996e264411f031/ui/components/MesheryMeshInterface/PatternServiceForm.js#L9)). Still WIP to make it respect light/dark mode theme (which is defined [here](https://github.com/meshery/meshery/blob/master/ui/themes/app.js) I suppose?).
> 2. For the displacement of `Settings` `AppBar` compenent, it seems it is caused by the disappearance of a MaterialUI CSS rule for this element class. The rules will reappear if we uncollapse and collapse back the `Settings` `AppBar`.
> 
> Currently I am able to tackle (1) (as seen on video above), but still figuring out why the CSS rule went missing on (2). However, as the `Settings` background color is also ruled by the same rule as (2), (1) is blocked by (2).

yes it is hard coded for which you can use globally provided `theme` to make dark mode compatible

**Comment by Yashsharma1911** [comment]
> For the `Settings` background color, is it supposed to have different color between light and dark theme? I could not find the design for dark theme in Figma.
> 
> ![image](https://user-images.githubusercontent.com/57625126/223633551-2ec3e738-b2e9-48bd-a061-da031fdbeb05.png)

Yes currently dark mode design is not provided but for color selection you can refer to this link [figma](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI?node-id=10319-32865&t=gXdm8ch3VCvm5Pul-4)

**Comment by Yashsharma1911** [comment]
> While investigating this ticket, I found another issue (although it seems it is unrelated to this ticket):
> 
> 1. Whenever the `Application` `LazyPatternServiceForm` is opened, the CSS rule of `body` will changed, which changed `font`-related rules (visible on `> Designs > New Pattern` breadcrumb in video below).
> 
> I still couldn't figure out how the theming switch works 😅

nice eye @mangkoran it would be nice if you can fix it, about theme switching if you are using makeStyle() provided by MUI then theme prop is provided which you can use or you can also use useTheme() hook by MUI to get current theme

**Comment by Yashsharma1911** [comment]
@mangkoran any progress?

**Comment by mangkoran** [comment]
Apologize, but I still stuck on how the theming works, which caused this issue.
> when switching between dark mode and light mode text settings is displaced from its place

I will create a draft PR for the "Settings" header background dark theme color. It seems we need additional help for the 2nd issue 😅

**Comment by Gauravjadhav22** [comment]
hey @Yashsharma1911, 
I want to work on this issue, is anyone still working on this or it has resolved?

**Comment by Yashsharma1911** [comment]
@Gauravjadhav22 are you able to run Meshery locally?

**Comment by Gauravjadhav22** [comment]
nope trying to set it up still stuck from few days :(

**Comment by Gauravjadhav22** [comment]
> @Gauravjadhav22 are you able to run Meshery locally?
nope trying to set it up still stuck from few days :(

**Comment by Gauravjadhav22** [comment]
nope trying to set it up still stuck from few days :(
please help ...

**Comment by Yashsharma1911** [comment]
@Gauravjadhav22 create post on [discuss.meshery.io](https://discuss.meshery.io/) if you are getting any issue so that community can help you and engage with you
